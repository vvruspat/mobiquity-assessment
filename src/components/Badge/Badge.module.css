.badge {
}

.badge_s {
    border-radius: 4px;
    padding: var(--padding-tiny) var(--padding-small);
}

.badge_m {
    border-radius: 4px;
    padding: var(--padding-extrasmall) var(--padding-regular);
}

.badge_l {
    border-radius: 8px;
    padding: var(--padding-regular) var(--padding-regular);
}

.badge_style_primary {
    background: var(--background-dark-color);
    color: var(--secondary-color);
}

.badge_style_secondary {
    background: var(--background-lightest-color);
    color: var(--regular-color);
}

.badge_style_info {
    background: var(--background-light-color);
    color: var(--regular-color);
}

.badge_style_accent {
    background: var(--accent-color);
    color: var(--regular-color);
}

.badge_align_left {
    text-align: left;
}

.badge_align_right {
    text-align: right;
}

.badge_align_center {
    text-align: center;
}