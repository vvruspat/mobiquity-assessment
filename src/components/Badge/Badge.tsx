import { FC, PropsWithChildren } from "react";
import cn from "clsx";

import s from "./Badge.module.css";

type BadgeProps = {
  /** Badge size affects the badge look */
  size?: "s" | "m" | "l";
  /** Text align */
  align?: "left" | "center" | "right";
  /** Badge style */
  badgeStyle?: "primary" | "secondary" | "info" | "accent";
};

/**
 * Badge component
 */
export const Badge: FC<PropsWithChildren<BadgeProps>> = ({
  size = "s",
  badgeStyle = "",
  align = "center",
  children,
}) => {
  return (
    <div
      className={cn(
        s.badge,
        s[`badge_${size}`],
        s[`badge_style_${badgeStyle}`],
        s[`badge_align_${align}`]
      )}
    >
      {children}
    </div>
  );
};
