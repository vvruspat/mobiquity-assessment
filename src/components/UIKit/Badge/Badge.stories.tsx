import { ComponentStory, ComponentMeta } from "@storybook/react";

import { Badge } from "./Badge";
import "../../../index.css";

export default {
  title: "UI Kit/Badge",
  component: Badge,
} as ComponentMeta<typeof Badge>;

const Template: ComponentStory<typeof Badge> = (args) => (
  <Badge {...args}>Badge content</Badge>
);

export const BadgeTemplate = Template.bind({});

BadgeTemplate.args = {
  align: "center",
  size: "s",
  badgeStyle: "secondary",
};
