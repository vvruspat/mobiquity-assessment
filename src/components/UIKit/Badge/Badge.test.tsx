import { render, screen } from "@testing-library/react";
import { Badge } from "./Badge";

test("renders badge", () => {
  render(<Badge>Test badge</Badge>);

  const el = screen.getByTestId("badge");

  expect(el).toHaveClass("badge_s");
  expect(el).toHaveClass("badge_align_center");
  expect(el).toMatchSnapshot();
});

test('renders badge size "m"', () => {
  render(<Badge size="m">Test badge</Badge>);

  const el = screen.getByTestId("badge");

  expect(el).toHaveClass("badge_m");
  expect(el).toHaveClass("badge_align_center");
  expect(el).toMatchSnapshot();
});

test('renders badge size "l"', () => {
  render(<Badge size="l">Test badge</Badge>);

  const el = screen.getByTestId("badge");

  expect(el).toHaveClass("badge_l");
  expect(el).toHaveClass("badge_align_center");
  expect(el).toMatchSnapshot();
});

test("renders badge align", () => {
  render(<Badge align="center">Test badge</Badge>);

  const el = screen.getByTestId("badge");

  expect(el).toHaveClass("badge_s");
  expect(el).toHaveClass("badge_align_center");
  expect(el).toMatchSnapshot();
});

test("renders badge align left", () => {
  render(<Badge align="left">Test badge</Badge>);

  const el = screen.getByTestId("badge");

  expect(el).toHaveClass("badge_s");
  expect(el).toHaveClass("badge_align_left");
  expect(el).toMatchSnapshot();
});

test("renders badge align right", () => {
  render(<Badge align="right">Test badge</Badge>);

  const el = screen.getByTestId("badge");

  expect(el).toHaveClass("badge_s");
  expect(el).toHaveClass("badge_align_right");
  expect(el).toMatchSnapshot();
});

test("renders badge accent", () => {
  render(<Badge badgeStyle="accent">Test badge</Badge>);

  const el = screen.getByTestId("badge");

  expect(el).toHaveClass("badge_s");
  expect(el).toHaveClass("badge_align_center");
  expect(el).toHaveClass("badge_style_accent");
  expect(el).toMatchSnapshot();
});

test("renders badge info", () => {
  render(<Badge badgeStyle="info">Test badge</Badge>);

  const el = screen.getByTestId("badge");

  expect(el).toHaveClass("badge_s");
  expect(el).toHaveClass("badge_align_center");
  expect(el).toHaveClass("badge_style_info");
  expect(el).toMatchSnapshot();
});

test("renders badge primary", () => {
  render(<Badge badgeStyle="primary">Test badge</Badge>);

  const el = screen.getByTestId("badge");

  expect(el).toHaveClass("badge_s");
  expect(el).toHaveClass("badge_align_center");
  expect(el).toHaveClass("badge_style_primary");
  expect(el).toMatchSnapshot();
});

test("renders badge secondary", () => {
  render(<Badge badgeStyle="secondary">Test badge</Badge>);

  const el = screen.getByTestId("badge");

  expect(el).toHaveClass("badge_s");
  expect(el).toHaveClass("badge_align_center");
  expect(el).toHaveClass("badge_style_secondary");
  expect(el).toMatchSnapshot();
});
