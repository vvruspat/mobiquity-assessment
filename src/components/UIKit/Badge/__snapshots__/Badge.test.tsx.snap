// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders badge 1`] = `
<div
  class="badge badge_s badge_align_center"
  data-testid="badge"
>
  Test badge
</div>
`;

exports[`renders badge accent 1`] = `
<div
  class="badge badge_s badge_style_accent badge_align_center"
  data-testid="badge"
>
  Test badge
</div>
`;

exports[`renders badge align 1`] = `
<div
  class="badge badge_s badge_align_center"
  data-testid="badge"
>
  Test badge
</div>
`;

exports[`renders badge align left 1`] = `
<div
  class="badge badge_s badge_align_left"
  data-testid="badge"
>
  Test badge
</div>
`;

exports[`renders badge align right 1`] = `
<div
  class="badge badge_s badge_align_right"
  data-testid="badge"
>
  Test badge
</div>
`;

exports[`renders badge info 1`] = `
<div
  class="badge badge_s badge_style_info badge_align_center"
  data-testid="badge"
>
  Test badge
</div>
`;

exports[`renders badge primary 1`] = `
<div
  class="badge badge_s badge_style_primary badge_align_center"
  data-testid="badge"
>
  Test badge
</div>
`;

exports[`renders badge secondary 1`] = `
<div
  class="badge badge_s badge_style_secondary badge_align_center"
  data-testid="badge"
>
  Test badge
</div>
`;

exports[`renders badge size "l" 1`] = `
<div
  class="badge badge_l badge_align_center"
  data-testid="badge"
>
  Test badge
</div>
`;

exports[`renders badge size "m" 1`] = `
<div
  class="badge badge_m badge_align_center"
  data-testid="badge"
>
  Test badge
</div>
`;
